### ISSUE TITLE
Suggestion: shells should exit only on Ctrl-D instead of Ctrl-C

### ISSUE THREAD

**Issue by martinvonwittich** [cli-guidelines/cli-guidelines#58]
From https://clig.dev/#signals:

> If a user hits Ctrl-C (the INT signal), exit as soon as possible.

I would suggest that this guideline should be split in two parts:

- If you're writing a shell (something that runs in a read-eval-print loop, for example a Unix shell like `bash` or a database client like `mysql`/`psql`), Ctrl-C should only interrupt the currently running command (if any), not the shell itself. If no command is currently running, Ctrl-C should discard any non-executed input received so far, so that a new command can be input. The shell itself should only quit when it receives Ctrl-D while it is waiting for input.

- If you write anything other than a shell, it should quit when receiving Ctrl-C.

**Comment by martinvonwittich** [comment]
This is something that I really hate about `mysql` and `asterisk -r` because they got it wrong :(

**Comment by aanand** [comment]
I agree - conceptually, to me at least, Ctrl-C means "cancel the current operation". Usually the "current operation" means the entire program, but in an interactive shell it means "the command I just invoked".

Interactive shells might even warrant their own section - there's a few good practices I'd like to encourage there, such as ensuring history and line editing work as expected.
